src/libstore/derivations.cc

```diff
@@ -103,31 +103,72 @@ StorePath writeDerivation(Store & store,
 }
 
 
-/* Read string `s' from stream `str'. */
-static void expect(std::istream & str, std::string_view s)
+/* Derivation parser local to this file.  Keep in unnabed namespace
+   to unblock aggressive inlining. */
+namespace {
+
+struct DerivationParserContext {
+    const char * ptr_;
+    const char * end_;
```

Comment:
Might be nicer to use an `std::string_view` here.

Reply:
`string_view` also directly carries the size instead of the end pointer, which could save a load and a compare to check for EOF in peek and get. might also make the scan loop in `parseString` a bit easierto read (using `find_first_of("\"\\")`)

Reply:
Sounds reasonable! Converted to `std::string_view`. The result is slightly more verbose in a few places, but `find_first_of()` is a lot less scary than initial version.